endpoints/openrtb2/auction.go

```diff
@@ -343,18 +330,54 @@ func (deps *endpointDeps) parseRequest(httpRequest *http.Request) (req *openrtb_
 
 	impInfo, errs := parseImpInfo(requestJson)
 	if len(errs) > 0 {
-		return nil, nil, nil, nil, nil, errs
+		return nil, nil, nil, nil, nil, nil, errs
+	}
+
+	storedBidRequestId, hasStoredBidRequest, storedRequests, storedImps, errs := deps.getStoredRequests(ctx, requestJson, impInfo)
+	if len(errs) > 0 {
+		return
+	}
+
+	accountId, isAppReq, errs := getAccountIdFromRawRequest(hasStoredBidRequest, storedRequests[storedBidRequestId], requestJson)
+	// fill labels here in order to pass correct metrics in case of errors
+	if isAppReq {
+		labels.Source = metrics.DemandApp
+		labels.RType = metrics.ReqTypeORTB2App
+		labels.PubID = accountId
+	} else { // is Site request
+		labels.Source = metrics.DemandWeb
+		labels.PubID = accountId
+	}
+	if errs != nil {
+		return
+	}
+
+	// Look up account
+	account, acctIDErrs := accountService.GetAccount(ctx, deps.cfg, deps.accounts, accountId)
+	if len(acctIDErrs) > 0 {
+		errs = append(errs, acctIDErrs...)
+		return
+	}
+
+	deps.hookExecutor.SetAccount(account)
+	requestJson, rejectErr = deps.hookExecutor.ExecuteRawAuctionStage(requestJson)
```

Comment:
The thing that worries me here, what if the module adds/modifies the stored request ID in the raw request? I can see the potential in an A/B testing module that substitutes one stored request for another randomly in order to get a sample of requests with the alternate configuration. Does the spec address this concern, or is it an open issue?